code/lib/builder-vite/src/plugins/externals-plugin.test.ts

```diff
@@ -0,0 +1,31 @@
+import { rewriteImport } from './externals-plugin';
+
+const packageName = '@storybook/package';
+const globals = { [packageName]: '_STORYBOOK_PACKAGE_' };
+
+const cases = [
+  {
+    globals,
+    packageName,
+    input: `import { Rain, Jour as Day, Nuit as Night, Sun } from "${packageName}"`,
+    output: `const { Rain, Jour: Day, Nuit: Night, Sun } = ${globals[packageName]}`,
+  },
+  {
+    globals,
+    packageName,
+    input: `import{Rain,Jour as Day,Nuit as Night,Sun}from '${packageName}'`,
+    output: `const {Rain,Jour: Day,Nuit: Night,Sun} = ${globals[packageName]}`,
+  },
+  {
+    globals,
+    packageName,
+    input: `const { Afternoon } = await import('${packageName}')`,
+    output: `const { Afternoon } = ${globals[packageName]}`,
+  },
```

Comment:
We should also handle `import * as foo from "package"`.  We might not have any of these right now, but we could encounter it in the future and we wouldn't want this to break.  It's maybe a bit trickier to handle, but we'd want it to become `const foo = __STORYBOOK_MODULE_PREVIEW_API__`.